File: test/merlintest2.py
import PyTango
import numpy
import time
import random

dev=PyTango.DeviceProxy('merlin/tango/1')
print "Software  version     :", dev.read_attribute("softwareversion").value

nframes = 50
exp_time = 0.01

lima=PyTango.DeviceProxy('limaccd/tango/3')
# do not change the order of the saving attributes!
# lima.write_attribute("saving_directory","/home/grm84/data")
# lima.write_attribute("saving_format","HDF5")
# lima.write_attribute("saving_overwrite_policy","Overwrite")
# lima.write_attribute("saving_suffix", ".hdf")
# lima.write_attribute("saving_prefix","merlin_")
# lima.write_attribute("saving_mode","AUTO_FRAME")
# lima.write_attribute("saving_frame_per_file", nframes)

dev.write_attribute("triggerStartType","INTERNAL");

for i in range(2):
    # do acquisition
    lima.write_attribute("acq_nb_frames",nframes)
    lima.write_attribute("latency_time",0.01)
    lima.write_attribute("acq_expo_time",exp_time)
    lima.write_attribute("acq_trigger_mode", "INTERNAL_TRIGGER")

    lima.command_inout("prepareAcq")
    start = time.time()
    lima.command_inout("startAcq")
    print "Running "
    while lima.read_attribute("acq_status").value == "Running" :
        time.sleep(.2)

    print "Completed run in ", time.time()-start, "secs"


print "Done"

File: test/MerlinTest2.cpp
/*
 * reply.cpp
 *
 *  Created on: Nov 26, 2014
 *      Author: grm84
 */
#include <iostream>
#include <string>
#include <sstream>
#include <iomanip>
#include <vector>
#include <unistd.h>
#include "lima/CtControl.h"
#include "lima/CtSaving.h"
#include "lima/CtAcquisition.h"
#include "lima/CtImage.h"
#include "lima/HwInterface.h"
#include "MerlinCamera.h"
#include "MerlinInterface.h"

using namespace std;
using namespace lima;
using namespace lima::Merlin;

DEB_GLOBAL(DebModTest);

int main () {
	DEB_GLOBAL_FUNCT();

	Camera *m_camera;
	Interface *m_interface;
	CtControl* m_control;

//	string hostname = "192.168.0.62";
    string hostname = "148.79.215.54";
	int port = 6341;
	int dataPort = 6342;
	int nx = 512;
	int ny = 512;
	int nchips = 4;
	bool simulate = false;

	try {

	  m_camera = new Camera(hostname, port, dataPort, nx, ny, nchips, simulate);
		m_interface = new Interface(*m_camera);
		m_control = new CtControl(m_interface);

		string type;
		m_camera->getDetectorType(type);
		cout << "Detector type : " << type << endl;
		string model;
		m_camera->getDetectorModel(model);
		cout << "Detector model: " << model << endl;
		float version;
		m_camera->getSoftwareVersion(version);
		cout << "Software version: " << version << endl;

		HwInterface::StatusType status;
		m_interface->getStatus(status);
		cout << "Detector Status: " << status << endl;

		//reset parameters for acquisition
		int nframes = 25;
		double time = 0.1;
		//		m_camera->setEnableCounters(Camera::COUNTER0);
		m_camera->setEnableCounters(Camera::BOTH);
		m_camera->setFramesPerTrigger(1);
		m_camera->setTriggerStartType(Camera::INTERNAL);
		m_camera->setTriggerStopType(Camera::INTERNAL);
		
		// setup fileformat and data saving info
		CtSaving* saving = m_control->saving();
		saving->setDirectory("/home/grm84/data");
		saving->setFramesPerFile(nframes);
		saving->setFormat(CtSaving::HDF5);
	 	saving->setPrefix("merlin_");
		saving->setSuffix(".hdf");
		saving->setSavingMode(CtSaving::AutoFrame);
		saving->setOverwritePolicy(CtSaving::Overwrite);

		// do acquisition
		m_control->image()->setImageType(lima::Bpp12);
		m_control->acquisition()->setAcqExpoTime(time);
		m_control->acquisition()->setLatencyTime(0);
		m_control->acquisition()->setAcqNbFrames(nframes);
		m_control->prepareAcq();
		m_control->startAcq();
		//		usleep(500000);
		//		m_control->stopAcq();
		sleep(5);

	} catch (Exception &e) {

	}
}
